src/main/java/spongebob/task/Deadline.java

```diff
@@ -0,0 +1,73 @@
+package spongebob.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import spongebob.exception.SpongebobException;
+
+/**
+ * A deadline task, contains a dateTime for due date
+ */
+public class Deadline extends Task {
+
+    protected LocalDate by;
```

Comment:
I think "by" may be slightly ambiguous though I'm aware this is the deadline date. Perhaps "deadline" might be a slightly more suitable variable name as it is a noun?

Reply:
Yep, I will change that too.